docs/operator-scoping.md

```diff
@@ -0,0 +1,148 @@
+
+OLM introduced the concept of `OperatorGroups` to enable cluster admins complete control over the permissions that OLM grants operators that it deploys. An admin may create a single `OperatorGroup` in a given namespace. Any CSV created in that namespaces is said to be a member operator of that `OperatorGroup`. With `OperatorGroups`, a cluster admin can:
+
+* Define the set of permissions that OLM may grant to member operators
+* Define the set of namespaces that OLM may grant namespaced permissions in.
+
+## Configuring OperatorGroups
+
+### Scoping Member Operators to Specific Namespaces
+
+Using an `OperatorGroup`, a cluster admin can scope member operators' namespaced permissions to specific namespaces in two way:
+
+* Using a predefined list of namespaces
+* Using a label selector.
+
+#### Defining a set of namespaces
+
+The set of namespaces can be hardcoded setting the `spec.targetNamespaces` of an `OperatorGroup` like so:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-group
+  namespace: my-namespace
+spec:
+  targetNamespaces:
+  - my-namespace
+  - my-other-namespace
+  - my-other-other-namespace
+```
+
+In the example above, member operator will be scoped to the following namespaces:
+
+* my-namespace
+* my-other-namespace
+* my-other-other-namespace
+
+#### Defining a set of namespaces with a label selector
+
+Cluster admins may not know in advance which namespaces member operators should be scoped to. In this case, it may make sense to use a label selector to identify which namespaces permissions should be granded in. A namespace selector can be defined for an Opearator group like so:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-group
+  namespace: my-namespace
+spec:
+  selector:
+    cool.io/prod: "true"
+```
+
+In the example above, member operators will be scoped to any namespaces with the `cool.io/prod: true` label.
+
+>Note: In the case that both a selector and a list of namespaces are provided, the selector is ignored.
+
+#### TargetNamespaces and their realationship to InstallModes
+
+When creating `OperatorGroups` it is important to keep in mind that an operator may not support all namespace configurations. For example, an operator that is designed to run at the cluster level shouldn't be expected to work in an `OperatorGroup` that defines a single targetNamespace.
+
+Operator authors are responsible for defining which `InstallModes` their operator supports within its `ClusterServiceVersion (CSV)`. An InstallMode consists of an InstallModeType field and a boolean Supported field. A CSV's spec can contain a set of InstallModes of four distinct InstallModeTypes:
+
+* **OwnNamespace**: If supported, the operator can be a member of an OperatorGroup that selects its own namespace
```

Comment:
The explanation around "operator can be a member of an OperatorGroup that selects..."  is too scientific. What it really comes down is that the Operator supports / does not support watching

- only the namespace it is deployed in
- a single given namespace
- a comma-separated list of namespaces
- the entire cluster

This paired with the OperatorGroup decides

- which namespace(s) the Operator is configured to watch
- in which namespace(s) the Operator is discoverable / useable by the cluster users

Reply:
Okay - I will update this portion of the PR to look similar to:
```
Operator authors are responsible for defining which `InstallModes` their operator supports within its `ClusterServiceVersion (CSV)`. There are four `InstallModes` that an operator can support:

* **OwnNamespace**: If supported, the operator can be configured to watch for events in the namespace it is deployed in.
* **SingleNamespace**:If supported, the operator can be configured to watch for events in a single namespace.
* **MultiNamespace**: If supported, the operator can be configured to watch for events in more than one namespace.
* **AllNamespaces**: If supported, the operator can be configured to watch for events in all namespaces.
```

Reply:
This has been resolved in the latest version of the commit - please take a moment to review.